cli/cli.js

```diff
@@ -194,3 +203,65 @@ function createJson(plans) {
         2,
     );
 }
+
+/**
+ * Runs the CLI Encode command
+ * @param {object} localOptions
+ * @param {Command} command
+ * @returns {Promise<void>}
+ */
+async function encode(localOptions, command) {
+    const globalParentOptions = command.parent.opts();
+    const { rpcUrl, fromBlock, pauseAddress } = globalParentOptions;
+
+    if (rpcUrl === defaults.RPC_URL) {
+        console.warn(
+            chalk.yellow(
+                `🟡 WARNING: Falling back to a public provider: ${rpcUrl}. For a better experience, set a custom RPC URL with the --rpc-url <rpc-url> option or the ETH_RPC_URL env variable.`,
+            ),
+        );
+    }
+
+    const spinner = ttyOnlySpinner().start("Fetching pending pause plans...");
+
+    try {
+        const pause = new ethers.Contract(
+            pauseAddress,
+            defaults.MCD_PAUSE_ABI,
+            ethers.getDefaultProvider(rpcUrl),
+        );
+
+        const status = "PENDING";
+        const plans = await fetchPausePlans(pause, { fromBlock, status });
+        spinner.success("Done!");
+
+        if (plans.length === 0) {
+            console.log(chalk.yellow("No pending spells found to encode."));
+            return;
+        }
+
+        const response = await prompts([
+            {
+                type: "multiselect",
+                name: "plans",
+                message:
+                    "Select spells to be encoded for `drop(Plan[] calldata _plans)`",
+                choices: plans.map((plan) => ({
+                    title: `hash: ${plan.hash} | usr: ${plan.usr} | eta: ${plan.eta}`,
+                    value: plan.hash,
+                })),
+            },
+        ]);
+
+        const selectedPlans = plans
+            .filter((plan) => response.plans.includes(plan.hash))
+            .map((plan) => [plan.usr, plan.tag, plan.fax, plan.eta.toString()]);
+
+        console.log("\n Encoded plans:");
+        console.log(chalk.green(JSON.stringify(selectedPlans)));
```

Comment:
Instead of manually encoding `plan.eta` here, we can simply leverage `createJSON` above, which already handles `BigInt`s.

There's only a small refactor required on [L203](https://github.com/makerdao/protego/pull/15/files#diff-54a0aa70bbfc306c7e92f0c21a52bbd6b76f53dd70e4d87714530195e58bab79R203). The hardcoded `2` should be extracted to a parameter `spaces` with default value `0`.
Then we need to update the call to it in the list command to pass `2` as the second parameter.

```suggestion
            .map((plan) => [plan.usr, plan.tag, plan.fax, plan.eta]);

        console.log("\n Encoded plans:");
        console.log(chalk.green(createJson(selectedPlans)));
```

Reply:
Done in [refactor: use createJson on encode command](https://github.com/makerdao/protego/pull/15/commits/808db10ec93596be428051cac0aa87ed70da073b).